client/src/pages/JournalPage.jsx

```
@@ -21,52 +22,60 @@ function JournalPage() {
     fetchEntries();
   }, [navigate]);
 
-  // Journal Logic
-  const fetchEntries = async () => {
-    try {
-      const response = await axios.get('http://localhost:3001/api/journal');
-      setEntries(response.data);
-      setLoading(false);
-    } catch (error) {
-      console.error('Error fetching entries:', error);
-      if (error.response?.status === 401) {
-        localStorage.removeItem('token');
-        navigate('/');
-      }
-    }
+  const fetchEntries = () => {
+    axios
+      .get('/api/journal')
+      .then((response) => {
+        if (response.status === STATUS.SUCCESS) {
+          setEntries(response.data);
+          setLoading(false);
+        }
+      })
+      .catch((error) => {
+        console.error('Error fetching entries:', error);
+        if (error.response?.status === STATUS.NOT_AUTHORIZED) {
+          localStorage.removeItem('token');
+          navigate('/');
+        }
+      });
   };
 
-  const handleSubmit = async (entryData) => {
-    console.log("Sending entry data:", entryData); // For debugging purposes
+  const handleSubmit = (entryData) => {
+    const request = editingEntry
+      ? axios.put(`/api/journal/${editingEntry.id}`, entryData)
+      : axios.post('/api/journal', entryData);
 
-    try {
-      if (editingEntry) {
-        await axios.put(`http://localhost:3001/api/journal/${editingEntry.id}`, entryData);
-        setEditingEntry(null);
-      } else {
-        await axios.post('http://localhost:3001/api/journal', entryData);
-      }
-      fetchEntries();
-    } catch (error) {
-      console.error('Error saving entry:', error);
-      alert('Failed to save entry. Please try again.');
-    }
+    request
+      .then((response) => {
+        if (response.status === STATUS.SUCCESS) {
```

should we throw an error if we get to this state and somehow the status is not success?